src/main/java/com/google/sps/DataServlet.java

```diff
@@ -187,7 +187,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     // The searched node exists but is not mutated in the current graph
     if (truncatedGraph.nodes().size() != 0
-        && mutationNumber != -1
+        && !(mutationNumber == -1 && nodeNameParam.equals(""))
```

Comment:
do you think it could make sense to separate this into 2 separate statements? I don't think the state of mutationNumber and nodeNameParam are necessarily linked?

Reply:
If mutationNumber is -1 and nodeNameParam is not empty then this case should always happen because a real node is never mutated in the initial graph. The only reason I had that check is to deal with the case when we are at the initial graph but we are not searching for a node. In that case no error message should be given. How do you think we should separate it?

Reply:
Or I guess it would be something like `mutationNumber != -1 || !nodeNameParam.equals(""))` if I'm distributing correctly. But i think we stuck with this in later branches so it's fine to keep!